gpii/node_modules/dbOperation/src/DbDataStoreUtils.js

```diff
@@ -0,0 +1,560 @@
+/*!
+GPII CouchDB/PouchDB Data Store Utilities
+
+Copyright 2016-2017 OCAD university
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    $ = fluid.registerNamespace("jQuery"),
+    uuid = uuid || require("node-uuid");
+
+fluid.registerNamespace("gpii.dbOperation.dbDataStore");
+
+/**
+ * Use the kettle dataSource `get` method to retrieve one record. This function provides extra
+ * verification on input required fields. It returns an empty object if the record is not found.
+ * This requires further processing besides using the kettle dataSource `notFoundIsEmpty` option because
+ * when retrieving CouchDB using views , an empty `rows` array rather than 404 (not found) http
+ * response code will be received when the record is not found.
+ * @param dataSource {Component} An instance of gpii.dbOperation.dbDataSource
```

Comment:
This doesn't match our current JSDocs conventions (and linting rules) - should read
@param {Component} dataSource - An instance
etc.

Reply:
I will update all these comments across the code base. 

BTW, according to [JSdoc comments enforced by ESLint](https://eslint.org/docs/rules/valid-jsdoc), there's no `-`(dash) in between the parameter name and its description. Just for us to reach a consensus.

Reply:
Got it. Adding hyphen.

Reply:
In most cases I was able to standardise the type ordering and dashes using "simple" regular expressions, hopefully it's not too involved.

I was just reading [the related ESlint docs](https://eslint.org/docs/rules/valid-jsdoc#options), there is a `matchDescription` option that we might set to something like `- .+` to both require a description and require it to start with a dash.  We could use this to make the convention required if it would make it easier to be consistent.  Happy to put together a pull on that if there is consensus that it's useful.

Reply:
I think it's a good idea to reinforce the convention with the linting

Reply:
Reading further, it looks like you can only specify a pattern for the overall description, and not the description for parameters or return values.  In reading through [this related ticket](https://github.com/eslint/eslint/issues/9827), it looks like:

1. There is an [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-require-hyphen-before-param-description) plugin that includes many checks we already have, but also hyphen checking.
2.  Although some of the plugin rules are already in core, this seems to have been brought in before there was a mature plugin infrastructure.  Although they won't remove existing options, the maintainers seem opposed to merging further rules from the plugin into their codebase.

So, in short, to do this we'd need to use an ESLint plugin and add that to our standard configuration.  Assuming that we could bring in the required plugin and configuration via the fluid-eslint-config project, it would only add one secondhand dependency.  Happy to test once we get input from @amb26 on the saneness/desirability of making the change.

Reply:
I don't feel it really pays its way to add a dependency to every project in order to lint a hyphen in our parameters. But let's sit on this use case and see if some other task emerges that we could achieve at the same time with such a plugin.